components/brave_wallet/renderer/brave_wallet_js_handler.cc

```diff
@@ -337,6 +336,7 @@ void BraveWalletJSHandler::OnRemoteDisconnect() {
 
 void BraveWalletJSHandler::AddJavaScriptObjectToFrame(
     v8::Local<v8::Context> context) {
+  EnsureConnected();
```

Comment:
This fixes things because `AddJavaScriptObjectToFrame` is only called when `brave_use_native_wallet` is true.